### PR TITLE
HSEARCH-4046 Expose a map of static children by name in the metamodel of object fields

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-4046

### CODE REVIEW

File: engine/src/main/java/org/hibernate/search/engine/backend/metamodel/IndexCompositeElementDescriptor.java [hunk 1]
```diff
@@ -44,8 +45,19 @@
 	 * fields created dynamically through {@link IndexSchemaElement#fieldTemplate(String, Function) templates}
 	 * are not included in the collection.
 	 *
-	 * @return A collection containing all fields.
+	 * @return A collection containing all static child fields.
 	 */
 	Collection<? extends IndexFieldDescriptor> staticChildren();
 
+	/**
+	 * Get all statically-defined, direct child fields for this element, mapped by their relative name.
```
Comment (fax4ever) [inline engine/src/main/java/org/hibernate/search/engine/backend/metamodel/IndexCompositeElementDescriptor.java:53]: Is relative name `IndexFieldDescriptor#relativeName`? Is would be not critical, but if so, maybe we could add the link to the method in the Javadoc.

**Review by fax4ever — APPROVED** [review]
Nice job. +1 to merge it

File: engine/src/main/java/org/hibernate/search/engine/backend/metamodel/IndexCompositeElementDescriptor.java (hunk 1 above)
Comment (yrodiere) [reply engine/src/main/java/org/hibernate/search/engine/backend/metamodel/IndexCompositeElementDescriptor.java]: Good idea. Done.

**Comment by yrodiere** [comment]
Merged, thanks!

**Pull request merged**